### DISCUSSION TITLE
Startschwierigkeiten und Fragen

### DISCUSSION THREAD

**Discussion by gjuecat** [wiggal/GEN24_Ladesteuerung#13] [Announcements]
Ich bekomme die Sache leider nicht zum Laufen.  Die Datei weatherData.json wurde zwar schon mal aktualisiert, allerdings mit keinen nutzbaren Daten:
`{
      "result": {
            "watts": {}
      },
      "messageCreated": "2023-04-20 15:33:02"`

Wenn ich start_LoggingSymoGen24.sh manuell ausführe, erhalte ich folgender Fehler:
`Traceback (most recent call last):
  File "/home/GEN24Ladesteuerung/LoggingSymoGen24.py", line 46, in <module>
    gen24 = SymoGen24Connector.SymoGen24(config['gen24']['hostNameOrIp'], config['gen24']['port'], auto)
  File "/home/GEN24Ladesteuerung/SymoGen24Connector.py", line 32, in __init__
    self.modbus.host(ipaddr)
TypeError: 'str' object is not callable
`

Wo starte ich nun am besten mit der Fehlersuche? Die config.ini habe ich durchgearbeitet und auf meine Daten angepasst.

Danke und lg

**Comment by wiggal** [comment]
Hi
1.)
Die Datei weatherData.json wird erzeugt, entweder durch
Solarprognose_WeatherData.py  Daten kommen dann von forecast.solar 
oder durch
Solarprognose_WeatherData.py  Daten kommen dann von http://www.solarprognose.de hier ist ein Account erforderlich

Sieht jetzt vom Ergebnis her so aus als wäre die Datei von Solarprognose_WeatherData.py erzeugt worden,
es wurden aber keine Daten empfangen., sind folgende Werte (XXXXX) vom Account bei http://www.solarprognose.de eingetragen?
```
[solarprognose]
weatherfile = weatherData.json
accesstoken = XXXXXXXXXXXXXXXXXXXXXXXXXXXX
item = inverter
id = XXXXX
type = hourly
dataAgeMaxInMinutes = 100
WaitSec = XXX
```

Ansonsten können die Wetterdaten durch start_Solarprognose_WeatherData.py.sh manuell geholt werden.

2.)
LoggingSymoGen24 ist nur zweitrangig um Daten aufzuzeichen.
Versuch mal die start_SymoGen24Controller2.sh zu starten, die Prozedur steuert die Batterieladung,
fuktioniert aber nur, wenn Wetterdaten vorhanden sind 1.)

Sind in der config.ini folgende Werte eingetragen?
```
[gen24]
# IP Wechselrichter
hostNameOrIp=XXX.XXX.XXX.XXX
port=502
```

Ist am Wechselrichter **Modbus aktiviert**?

Bitte die Fragen abklären

Fehlermeldungen usw. bitte nach Eingabe immer markieren und mir "<>" im Menue als Code markieren,
dann ist das ganze leichter lesbar.

Schönen Gruß

**Comment by gjuecat** [comment]
Hallo,

die weatherData.json hat heute einen passenden Inhalt mit den stündlichen Werten.

config.ini enhält die Daten, Modbus ist auch aktiv (und funktioniert mit anderen Systemen):
```
[gen24]
# IP Wechselrichter
hostNameOrIp=192.168.36.210
port=502
```

Wenn ich jedoch die start_SymoGen24Controller2.sh ausführe, kommt folgende Meldung:

```
root@s36-debian-1:/home/GEN24Ladesteuerung# ./start_SymoGen24Controller2.sh
  Traceback (most recent call last):
  File "/home/GEN24Ladesteuerung/SymoGen24Controller2.py", line 212, in <module>
    gen24 = SymoGen24Connector.SymoGen24(config['gen24']['hostNameOrIp'], config['gen24']['port'], auto)
  File "/home/GEN24Ladesteuerung/SymoGen24Connector.py", line 32, in __init__
    self.modbus.host(ipaddr)
TypeError: 'str' object is not callable
```

Danke und lg

**Comment by wiggal** [comment]
Schein ein Problem vom pyModbusTCP Modul zu sein.
Poste mal die Ausgabe von:
pip show pyModbusTCP 
sollte volgende Version haben:
Name: pyModbusTCP

```
Version: 0.1.10
Summary: A simple Modbus/TCP library for Python
Home-page: https://github.com/sourceperl/pyModbusTCP
Author: Loic Lefebvre
Author-email: loic.celine@free.fr
License: MIT
Location: /usr/local/lib/python3.9/dist-packages
Requires: 
Required-by: 
```
lg

**Comment by gjuecat** [comment]
da hab ich 0.2.0

```
root@s36-debian-1:/home/GEN24Ladesteuerung# pip show pyModbusTCP
Name: pyModbusTCP
Version: 0.2.0
Summary: A simple Modbus/TCP library for Python
Home-page: https://github.com/sourceperl/pyModbusTCP
Author: Loic Lefebvre
Author-email: loic.celine@free.fr
License: MIT
Location: /usr/local/lib/python3.9/dist-packages
Requires:
Required-by:
```

bin nun zurück auf 0.1.10 und dann läuft start_SymoGen24Controller2.sh ohne Fehlermeldung durch

lg

**Comment by wiggal** [comment]
OK habs probiert, liegt an der Version: 0.2.0,
wird etwas dauern, bis ich das Ganze auf den neuen Stand bringe.

Wenn du willst, könntest du einen Downgrade versuchen:

 pip install pyModbusTCP==0.1.10

lg

**Comment by gjuecat** [comment]
ja, ich bin eh schon auf 0.1.10 zurück, dann ist der Fehler weg.

Mein Akku steht jetzt auf 70% und wird nicht mehr geladen. Lt. Config.ini soll er um 15Uhr voll sein. Wie bekomme ich nun den Akku bis zu Abend voll?

**Comment by gjuecat** [comment]
Vielleicht passts nun eh, er hat nun zumindest angefangen zu laden.

**Comment by wiggal** [comment]
wenn der Akku nach config um 15:00 voll sein soll und er warum auch immer (evtl. zu wenig Grundlast konfiguriert, oder Prognose besser als Wetter ), dann schaltet er auf volle Ladung (MaxLadung). Wenn er um 14:45 erst 70% hat, müsste er auch schon auf MaxLadung stehen. Dann muss nur noch das Wetter mitspielen damit er noch voll wird. ;-)
lg

**Comment by wiggal** [comment]
Hi
schließe hier mit erledigt.
Weitere Fragen gerne aber bitte unter Discusions.
viel Erfolg mit dem Tool
